specs/OAuth/AuthManager.md

```diff
@@ -0,0 +1,891 @@
+AuthManager API
```

Comment:
I'm not loving the "Auth" prefix to every class. I'd spell it out `Authentication*`.
As a static class, I'm also not quite clear on what makes this a "Manager".

Reply:
OAuthentication is another thought for the class name. Let me know your views.

Reply:
"OAuth 2.0 is generally more popular and widely adopted for authorization purposes. It is used by major platforms like Google, Facebook, and GitHub to enable secure data sharing between applications without exposing user credentials67. Web Authentication, on the other hand, is a broader term that encompasses various methods of authenticating users on the web, including OAuth 2.0. While OAuth 2.0 is specifically designed for authorization, Web Authentication can include other protocols and methods such as Basic Authentication, JWT (JSON Web Tokens), and more"

It is popular enough, this is the response I'm getting from chatGPT :), So we are going ahead with OAuth2Manager, and as you said, we are managing the flow of OAuth2 request.
Let us know if you see some issue with OAuth2Manager.

Reply:
Updated the name to "OAuth2Manager"